src/index_map.rs

```diff
@@ -0,0 +1,432 @@
+//! [`NEIndexMap`] is a non-empty variant of [`IndexMap`].
+#![deny(clippy::correctness, clippy::suspicious)]
+#![warn(clippy::complexity, clippy::perf, clippy::style, clippy::pedantic)]
+#![allow(clippy::module_name_repetitions)]
+use std::fmt::Debug;
+use std::hash::BuildHasher;
+use std::hash::Hash;
+use std::iter;
+use std::mem;
+
+use crate::FromNonEmptyIterator;
+use crate::IntoNonEmptyIterator;
+use crate::NonEmptyIterator;
+use indexmap::Equivalent;
+use indexmap::IndexMap;
+
+/// Short-hand for constructing [`NEIndexMap`] values.
+///
+/// ```
+/// use nonempty_collections::ne_indexmap;
+///
+/// let m = ne_indexmap!{"elves" => 3000, "orcs" => 10000};
+/// assert_eq!(2, m.len());
+/// ```
+#[macro_export]
+macro_rules! ne_indexmap {
```

Comment:
How about `nei`? It's otherwise somewhat verbose. I'm open to other suggestions.

Reply:
I wasn't aware that the macro `indexmap` already existed, so I'm okay with following your suggestion here. `nev` and friends will remain as-is, as:

- that matches the three-letter `vec!` from the standard library
- those namings are documented heavily here and hard to miss